Microsoft.Research/Clousot.Cache/SQLCacheModel.cs

```diff
@@ -784,5 +786,33 @@ protected override SqlConnectionStringBuilder BuildConnectionString(IClousotCach
 
       return base.BuildConnectionString(options);
     }
+
+    public static string GetDesiredDbName()
```

Comment:
So my point was: extracting this logic our of the factory clearly separates the concerns and responsibilities of the code.